superdesk/io/format_document_for_preview.py

```diff
@@ -0,0 +1,43 @@
+from flask import Blueprint, request, Response
+import superdesk
+from superdesk import get_resource_service
+from superdesk.publish.formatters import get_formatter
+from apps.content_types import apply_schema
+
+
+bp = Blueprint('format_document', __name__)
+
+
+def get_mime_type(formatter_qcode):
+    if(formatter_qcode == 'newsmlg2'):
+        return 'text/xml'
+    else:
+        return 'application/json'
+
+
+@bp.route('/format-document-for-preview/', methods=['GET', 'OPTIONS'])
+def format_document():
+
+    document_id = request.args.get('document_id')
+    subscriber_id = request.args.get('subscriber_id')
+    formatter_qcode = request.args.get('formatter')
+
+    subscriber = get_resource_service('subscribers').find_one(req=None, _id=subscriber_id)
+    doc = get_resource_service('archive').find_one(req=None, _id=document_id)
```

Comment:
imo we should send the item for preview in the payload, otherwise you would have to save before you can preview

Reply:
Let's require saving https://github.com/superdesk/superdesk-client-core/pull/2429#discussion_r203343483

The publisher team needs the same API as well, so using an ID works for everyone. We can add another endpoint with different parameters if saving before previewing proves to be inconvenient.